### PR TITLE
feat: [CI-4835]: add ACR step

### PR DESCRIPTION
**Pull request by d1wilko**
### Summary

Add ACR step to build steps

#### Screenshots

image

image

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by d1wilko** [comment]
retrigger titlecheck

File: src/modules/75-ci/components/PipelineSteps/ACRStep/ACRStepBase.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                      stepViewType={stepViewType}
+                      readonly={readonly}
+                      enableFields={{
+                        'spec.optimize': { shouldHide: buildInfrastructureType === 'VM' },
+                        'spec.dockerfile': {},
+                        'spec.context': {},
+                        'spec.labels': {},
+                        'spec.buildArgs': {},
+                        'spec.target': { tooltipId: 'target' },
+                        'spec.remoteCacheImage': { shouldHide: buildInfrastructureType === 'VM' }
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/ACRStep/ACRStepBase.tsx:126]: Better to use CIBuildInfrastructureType.VM from `75-ci/constants/Constants.ts`

File: package.json [hunk 2]
```diff
@@ -58,7 +58,7 @@
     "@emotion/styled": "^10.0.27",
     "@harness/design-system": "1.0.0",
     "@harness/help-panel": "1.8.0",
-    "@harness/icons": "^1.42.0",
+    "@harness/icons": "^1.43.0",
```
Comment (mtran7) [inline package.json:61]: ^1.42.0 should already include minor version 1.43.0 so change would need need to be required. Can test range with this semver calculator https://semver.npmjs.com/

File: src/modules/75-ci/components/PipelineSteps/ACRStep/ACRStepBase.tsx (hunk 1 above)
Comment (d1wilko) [reply src/modules/75-ci/components/PipelineSteps/ACRStep/ACRStepBase.tsx:126]: sure, good catch

File: package.json (hunk 2 above)
Comment (d1wilko) [reply package.json:61]: In theory you are right, but it doesn't seem to be working and people have been updating the package number to get it to work :(

**Review by vardanbansal-harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Pull request merged**